### PR TITLE
baseapp-ratings

### PR DESCRIPTION
**Pull request by vitorguima**

### CODE REVIEW

**Comment by vitorguima** [comment]
[] missing Readme

File: baseapp-ratings/baseapp_ratings/models.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    )
+
+    target_content_type = models.ForeignKey(
+        ContentType,
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        db_index=True,
+    )
+    target_object_id = models.PositiveIntegerField(blank=True, null=True, db_index=True)
```
Comment (nossila) [inline baseapp-ratings/baseapp_ratings/models.py:46]: I think we can remove the `db_index` here as well, I see no reason to filter only by `target_object_id` and not use `target_content_type` in the filter

**Review by nossila — CHANGES_REQUESTED** [review]

File: baseapp-auth/baseapp_auth/graphql/object_types.py [hunk 2]
```diff
@@ -24,6 +24,12 @@
     interfaces += (PageInterface,)
 
 
+if apps.is_installed("baseapp_ratings"):
+    from baseapp_ratings.graphql.object_types import RatingsInterface
+
+    interfaces += (RatingsInterface,)
```
[the remark this answers is not in the document]
Comment (vitorguima) [reply baseapp-auth/baseapp_auth/graphql/object_types.py:30]: yes

File: baseapp-ratings/baseapp_ratings/admin.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+class RatingAdmin(admin.ModelAdmin):
+    search_fields = (
+        "user__first_name",
+        "user__last_name",
+    )
+    raw_id_fields = ("user",)
+    list_display = (
+        "id",
+        "target",
+        "user",
```
[the remark this answers is not in the document]
Comment (vitorguima) [reply baseapp-ratings/baseapp_ratings/admin.py:17]: done

File: baseapp-ratings/baseapp_ratings/graphql/object_types.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            "user",
+            "created",
+            "modified",
+            "target",
+            "value",
+        )
+
+    @classmethod
+    def get_node(self, info, id):
+        node = super().get_node(info, id)
```
[the remark this answers is not in the document]
Comment (vitorguima) [reply baseapp-ratings/baseapp_ratings/graphql/object_types.py:63]: done

File: baseapp-ratings/baseapp_ratings/graphql/mutations.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        gid_type, gid = from_global_id(input.get("target_object_id"))
+        object_type = info.schema.get_type(gid_type)
+        target = object_type.graphene_type.get_node(info, gid)
+
+        target_content_type = input.get("target_content_type")
+        if target_content_type:
+            content_type = get_content_type_by_natural_key(target_content_type)
+            target = content_type.get_object_for_this_type(pk=gid)
+        else:
+            content_type = ContentType.objects.get_for_model(target)
```
[the remark this answers is not in the document]
Comment (vitorguima) [reply baseapp-ratings/baseapp_ratings/graphql/mutations.py]: done

File: baseapp-ratings/baseapp_ratings/graphql/object_types.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+RateModel = swapper.load_model("baseapp_ratings", "Rate")
+
+
+class RatingsInterface(relay.Node):
+    ratings_count = graphene.Int()
+    ratings_sum = graphene.Int()
+    ratings_average = graphene.Float()
+    ratings = DjangoConnectionField(lambda: RatingObjectType)
+    is_ratings_enabled = graphene.Boolean(required=True)
+    my_reaction = graphene.Field(lambda: RatingObjectType, required=False)
```
[the remark this answers is not in the document]
Comment (vitorguima) [reply baseapp-ratings/baseapp_ratings/graphql/object_types.py]: done

File: baseapp-ratings/baseapp_ratings/graphql/object_types.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+    def resolve_ratings(self, info, **kwargs):
+        if not getattr(self, "is_ratings_enabled", True):
+            return RateModel.objects.none()
+
+        CAN_ANONYMOUS_VIEW_REACTIONS = getattr(
+            settings, "BASEAPP_REACTIONS_CAN_ANONYMOUS_VIEW_RATINGS", True
+        )
+        if not CAN_ANONYMOUS_VIEW_REACTIONS and not info.context.user.is_authenticated:
```
[the remark this answers is not in the document]
Comment (vitorguima) [reply baseapp-ratings/baseapp_ratings/graphql/object_types.py]: done

**Review by nossila — APPROVED** [review]

**Pull request merged**